src/modules/map/components/openlayers/OpenLayersGeoJSONLayer.vue

```diff
@@ -46,66 +46,58 @@ export default {
                 this.layer.setOpacity(newOpacity)
             }
         },
+        geojsonUrl() {
+            this.loadData()
+        },
+        styleUrl() {
+            this.loadStyle()
+        },
     },
     created() {
-        // The layers needs to be set immediately. Otherwise, addLayerToMapMixin will fail.
+        // The layer needs to be set immediately. Otherwise, addLayerToMapMixin will fail.
         this.layer = new VectorLayer({ id: this.layerId, opacity: this.opacity })
+    },
+    mounted() {
+        this.loadStyle()
+        this.loadData()
+    },
+    methods: {
+        async loadStyle() {
+            try {
+                const style = new OlStyleForPropertyValue((await axios.get(this.styleUrl)).data)
+
+                this.layer.setStyle(function (feature, res) {
+                    return style.getFeatureStyle(feature, res)
+                })
+            } catch (err) {
+                log.error('Unable to load style for GeoJSON layer', this.layerId, err)
+            }
```

Comment:
I would not put the try catch block around everything, but only around the axios call.
During my test the loadStyle() failed as follow:
```
Unable to load style for GeoJSON layer ch.bfe.ladestellen-elektromobilitaet TypeError: Cannot read properties of undefined (reading 'setStyle')
    at Proxy.loadStyle (index-373d0628.js:1410:17195)
```
The Axios call was successful which means that the data could not be understood by openlayer, I don't know if letting crash would provide more infos here but we clearly don't don't know why it happens.